### PR TITLE
Suggestions for LM API description

### PR DESCRIPTION
**Pull request by kikkomep**

### CODE REVIEW

File: lm_api_specs.md [hunk 1]
```diff
@@ -1,35 +1,47 @@
...
-  * must authenticate with API Key or OAuth AuthorizationCodeFlow;
-  * can't use Registry Client API resources.
-* **Registry clients**:
-  * must authenticate with OAuth RegistryClientCredentials or OAuth RegistryCodeFlow;
-  * can't use User Client API resources.
+* **API key**, can be used to implement generic clients;
+* **OAuth2 Client Credentials**, only available for trusted workflow registries (see *RegistryClientCredentials*);
+* **OAuth2 Authorization Code**, available for both generic clients (see *AuthorizationCodeFlow*) and registry clients (see *RegistryCodeFlow*).
 
+Clients can query API endpoints accordingly to the authorisation mechanism they adopt.
```
Comment (simleo) [inline lm_api_specs.md]: "accordingly to" -> "according to"

File: lm_api_specs.md [hunk 2]
```diff
@@ -1,35 +1,47 @@
...
+There are in fact resources which are _"contextual"_ to the actor which the client is acting on behalf of and the type of "impersonated" actor is determined by the authorisation grant in use. Examples of contextual resources are:
 
-Follow [this link to see the API documentation and examples](https://api-lifemonitor-dev.rahtiapp.fi/static/apidocs.html).  You can find the raw YAML API specifications in [the source code repository](https://github.com/crs4/life_monitor/tree/master/specs).  
+- `/users/current/*`, which assume an authenticated _user_ and thus can be queried only by clients authorised through an _API key_ or an _OAuth2 Authorization Code flow_;
 
+- `/registries/current/*`, which assume an authenticated _registry_ and thus can be queried only by clients authorised through an _API key_ or an _OAuth2 Authorization Code flow_.
```
Comment (simleo) [inline lm_api_specs.md]: I think you mean "authorised through _RegistryClientCredentials_ or _RegistryCodeFlow_" here

File: lm_api_specs.md [hunk 3]
```diff
@@ -1,35 +1,47 @@
...
-  * can't use User Client API resources.
+* **API key**, can be used to implement generic clients;
+* **OAuth2 Client Credentials**, only available for trusted workflow registries (see *RegistryClientCredentials*);
+* **OAuth2 Authorization Code**, available for both generic clients (see *AuthorizationCodeFlow*) and registry clients (see *RegistryCodeFlow*).
 
+Clients can query API endpoints accordingly to the authorisation mechanism they adopt.
+There are in fact resources which are _"contextual"_ to the actor which the client is acting on behalf of and the type of "impersonated" actor is determined by the authorisation grant in use. Examples of contextual resources are:
 
-Follow [this link to see the API documentation and examples](https://api-lifemonitor-dev.rahtiapp.fi/static/apidocs.html).  You can find the raw YAML API specifications in [the source code repository](https://github.com/crs4/life_monitor/tree/master/specs).  
+- `/users/current/*`, which assume an authenticated _user_ and thus can be queried only by clients authorised through an _API key_ or an _OAuth2 Authorization Code flow_;
```
Comment (simleo) [inline lm_api_specs.md]: Looking at the specs, I think that currently `/users/current` and `/users/current/workflows` are also accessible via RegistryCodeFlow

**Comment by kikkomep** [comment]
thanks @simleo

**Pull request merged**